gateway/api_test.go

```diff
@@ -2159,6 +2159,7 @@ func TestOAS(t *testing.T) {
 				oldAPIInOAS.GetTykExtension().Info.Versioning = nil
 				oldAPIInOAS.GetTykExtension().Server.GatewayTags = nil
 				oldAPIInOAS.GetTykExtension().Upstream.MutualTLS = nil
+				oldAPIInOAS.GetTykExtension().Upstream.CertificatePinning = nil
```

Comment:
I'd make a local var with the tyk extension to avoid the function call overhead.

Why do we need to clear each optional OAS setting to nil? Not part of ticket, but we should consider this a defect. Clearing each newly added *ptr field to nil in tests is just an indicator that we wrongly allocate the tyk extension if the default is to have these fields allocated...